common/pinot/pinotQueryValidator.go

```diff
@@ -328,17 +328,24 @@ func (qv *VisibilityQueryValidator) processCustomKey(expr sqlparser.Expr) (strin
 	if !ok {
 		return "", fmt.Errorf("invalid search attribute")
 	}
+	// get the value type
+	indexValType := common.ConvertIndexedValueTypeToInternalType(valType, log.NewNoop())
```

Comment:
let's move this line after get the colVal, so when it is invalid comparison expression, we don't need to do this conversion.